packages/format-library/src/link/inline.js

```diff
@@ -113,71 +113,64 @@ function InlineLinkUI( {
 
 		const newText = nextValue.title || newUrl;
 
+		// Scenario: we have any active text selection or an active format.
+		let newValue;
 		if ( isCollapsed( value ) && ! isActive ) {
 			// Scenario: we don't have any actively selected text or formats.
-			const toInsert = applyFormat(
-				create( { text: newText } ),
+			const inserted = insert( value, newText );
+
+			newValue = applyFormat(
+				inserted,
 				linkFormat,
-				0,
-				newText.length
+				value.start,
+				value.end + newText.length
```

Comment:
Isn't start and end same if isCollapsed?

Reply:
Yes it should be 🤔

Reply:
Yeah, they are the same. Not sure why I did it that way. Probably clearer to use value.start.